gobblin-compaction/src/main/java/org/apache/gobblin/compaction/mapreduce/orc/OrcKeyCompactorOutputFormat.java

```diff
@@ -39,4 +49,22 @@ public synchronized OutputCommitter getOutputCommitter(TaskAttemptContext contex
     }
     return this.committer;
   }
+
+  /**
+   * Required for extension since super method hard-coded file extension as ".orc". To keep flexibility
+   * of extension name, we made it configuration driven.
+   * @param taskAttemptContext The source of configuration that determines the file extension
+   * @return The {@link RecordWriter} that write out Orc object.
+   * @throws IOException
+   */
+  @Override
+  public RecordWriter getRecordWriter(TaskAttemptContext taskAttemptContext) throws IOException {
+    Configuration conf = taskAttemptContext.getConfiguration();
+    String extension = "." + conf.get(COMPACTION_OUTPUT_EXTENSION, "orc" );
```

Comment:
Do you think empty extension should be supported?

Reply:
That makes filtering of file format harder. I think we need to add some extension, hopefully consistent, to orc files to avoid checking bytes to determine if a file is ORC every time. 

Plus, we are less likely to run compaction on existing no-extension files.

Reply:
Okay.